lib/generators/graphql/type_generator.rb

```diff
@@ -82,15 +97,37 @@ def normalized_fields
         }
       end
 
+      def ruby_class_name
+        class_prefix = 
+          if options[:namespaced_types]
+            "#{graphql_type.pluralize.camelize}::"
+          else
+            ""
+          end
+        @ruby_class_name || class_prefix + type_ruby_name.sub(/^Types::/, "")
+      end
+
+      def subdirectory
+        if options[:namespaced_types]
+          "/#{graphql_type.pluralize}"
+        else
+          ""
+        end
+      end
+
       class NormalizedField
         def initialize(name, type_expr, null)
           @name = name
           @type_expr = type_expr
           @null = null
         end
 
-        def to_ruby
-          "field :#{@name}, #{@type_expr}, null: #{@null}"
+        def to_object_field
+          "field :#{@name}, #{@type_expr}#{@null ? '' : ', null: false'}"
+        end
+
+        def to_input_argument
+          "argument :#{@name}, #{@type_expr}, required: false"
         end
```

Comment:
How about considering`@null` when generating input arguments? 

```suggestion
        def to_input_argument
          "argument :#{@name}, #{@type_expr}#{@null ? ? ", required: false" : ""}"
        end
```

Reply:
I considered it, but ultimately decided against it.

When working on my own projects I tried to make non-nullable fields required myself, but although it worked wonderfully when creating new records, it quickly broke down when editing them. It forced me to send back a value for every non-nullable field, even if I wasn't updating it, so in the end I found it more expedient to make all arguments for these sort of inputs optional, and turn database/model errors from required fields into `GraphQL::ExecutionError`s

Reply:
Fair enough, thanks for sharing your rationale! If later users report that the generator is "ignoring" their requests to make `required: true` input fields, I may update this behavior in the future, but we'll see.